### PR TITLE
🚸 Self-recover missing NFT sold info in DB

### PR DESCRIPTION
**Pull request by WeiJun0827**

### CODE REVIEW

File: src/util/api/likernft/purchase.ts [hunk 1]
```diff
@@ -346,6 +352,84 @@ export async function handleNFTPurchaseTransaction({
...
+    grantTxHash,
+    granterMemo,
+    memo,
+    price,
+    classId,
+    nftId,
+    timestamp,
+    fromWallet: seller,
+    toWallet: owner,
+    granterWallet,
```
Comment (WeiJun0827) [inline src/util/api/likernft/purchase.ts:431]: Note that we don't provide `sellerWallet, sellerLIKE, stakeholderWallets, stakeholderLIKEs` which appear in the successful purchase flow. Not sure if it's necessary for now/future.

File: src/util/api/likernft/purchase.ts [hunk 2]
```diff
@@ -553,7 +638,24 @@ export async function processNFTPurchase({
...
+          classRef,
+          classId,
+          nftId,
+          granterWallet,
+          grantTxHash,
+          granterMemo,
+        });
+      }
       t.update(nftRef, { isProcessing: false });
+      return shouldUpdateSoldNFT && retryTimes < 1;
```
Comment (WeiJun0827) [inline src/util/api/likernft/purchase.ts:660]: If the flow is acceptable, probably can move the `MAX_NFT_PURCHASE_RETRY_TIMES` to config.js

File: src/util/api/likernft/purchase.ts [hunk 3]
```diff
@@ -346,6 +352,84 @@ export async function handleNFTPurchaseTransaction({
...
+    'pagination.limit': 1,
+    'pagination.reverse': true,
+  };
+  const { data: { events } } = await axios.get(`${COSMOS_LCD_INDEXER_ENDPOINT}/likechain/likenft/v1/event`, { params });
+  return events.length ? events[0] : null;
+}
+
+function formatNFTEvent(event: any) {
+  const timestamp = new Date(event.timestamp).getTime();
+  const price = Number(new BigNumber(event.price).shiftedBy(-9).toFixed(0));
```
Comment (williamchong) [inline src/util/api/likernft/purchase.ts:372]: Why can we do `toFixed(0)` here? Price can be non-integer

**Review by williamchong — CHANGES_REQUESTED** [review]

File: src/util/api/likernft/purchase.ts (hunk 3 above)
Comment (WeiJun0827) [reply src/util/api/likernft/purchase.ts:372]: The `event.price` includes tx fee (e.g. `8002000000 nanolike`), but the price in FIrestore is an integer that excludes tx fee. Given the condition of NFT sold by API wallet, the tx price queried from `/event` API should be formatted to an integer after excluding the tx fee?

Comment (williamchong) [reply src/util/api/likernft/purchase.ts:372]: Please add a comment to explain this situation

**Review by nnkken — APPROVED** [review]

**Review by williamchong — APPROVED** [review]

**Pull request merged**